### PR TITLE
Parse <register><alternateRegister> elements

### PR DESCRIPTION
**Pull request by wez**
In looking at https://github.com/japaric/svd2rust/issues/191 and https://github.com/japaric/svd2rust/issues/16 I thought that this might help, so here's the trivial patch.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 *.rs.bk
+.*sw*
```
Comment (Emilgardis) [inline .gitignore]: `[._]*.sw[a-p]` is a better solution for matching vim swap files.

File: src/lib.rs [hunk 2]
```diff
@@ -351,6 +354,13 @@ impl RegisterInfo {
     fn parse(tree: &Element) -> RegisterInfo {
         RegisterInfo {
             name: try!(tree.get_child_text("name")),
+            alternate_group: tree.get_child_text("alternateGroup"),
+            alternate_register: tree.get_child_text("alternateRegister"),
+            derived_from: tree.attributes.get("derivedFrom").map(
+                |s| {
+                    s.to_owned()
+                },
+            ),
```
Comment (Emilgardis) [inline src/lib.rs]: This indentation is uneeded

```rust
tree.attributes.get("derivedFrom").map(|s| s.to_owned()),
```

File: .gitignore (hunk 1 above)
Comment (wez) [reply .gitignore]: will amend

File: src/lib.rs (hunk 2 above)
Comment (wez) [reply src/lib.rs]: I copied this from the other instance of `derivedFrom`.  Note that the `.rustfmt.toml` config wants this to be:

```
            derived_from: tree.attributes
                .get("derivedFrom")
                .map(|s| s.to_owned()),
```

There are other portions that are not rustfmt'd.   Would you like me to insert a commit that applies rustfmt underneath my changes and then make them match?

**Comment by wez** [comment]
https://github.com/japaric/svd/pull/51 for rustfmt, I'll amend this one independently from that for now

**Comment by wez** [comment]
(force pushed with your suggested changes; you'll probably need to hard refresh the PR page)

**Comment by Emilgardis** [comment]
Thank you!

bors r+

**Pull request merged**